ros2doctor/ros2doctor/api/package.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+from ament_index_python import get_packages_with_prefixes
+from catkin_pkg.package import InvalidPackage
+from catkin_pkg.package import parse_package
+from ros2doctor.api import DoctorCheck
+from ros2doctor.api import DoctorReport
+from ros2doctor.api import Report
+from ros2doctor.api import Result
+
+import rosdistro
+
+
+def get_ros_packages_info() -> dict:
+    """
+    Return all current distro's packages info using rosdistro API.
+
+    :return: a dictionary contains package name, release, source and status
+    """
+    distro_name = os.environ.get('ROS_DISTRO')
+    distro_name = distro_name.lower()
+    url = rosdistro.get_index_url()
+    i = rosdistro.get_index(url)
+    distro_data = rosdistro.get_distribution(i, distro_name).get_data()
+    return distro_data.get('repositories')
+
+
+class PackageCheck(DoctorCheck):
+    """Check local package versions against release versions on rosdistro."""
+
+    def category(self):
+        return 'package'
+
+    def check(self):
+        """Check packages within the directory where command is called."""
+        result = Result()
+        try:
+            distro_packages_info = get_ros_packages_info()
```

Comment:
Recommend `distro_packages_info -> distro_repositories_info` and `get_ros_packages_info -> get_ros_repositories_info` since it's a dictionary of repositories, which may have 1 or more packages in them.